### PR TITLE
Fallback to fix #19 bug

### LINKED ISSUE #19: [1.5.1] Broken when using defaultText
When a default prop is passed to the `Select` component in example here:
```js
<Select
  onSelect = {d => this.props.onChange({farm : d})}
  defaultText  = "Choose"
  {
    this.state.farms.map((farm, key) => 
      <Option value={farm.id} key={key}>{farm.name}</Option>
    )
  }
</Select>
```
This will render the input selected to defaultText and not the selected value.

Making a PR to fix it

### PR DESCRIPTION
**Pull request by st0ffern**
Closes: https://github.com/gs-akhan/react-native-chooser/issues/19

### CODE REVIEW

**Comment by st0ffern** [comment]
cc @gs-akhan

File: lib/select.js [hunk 1]
```diff
@@ -46,15 +46,7 @@ export default class Select extends Component {
 		  selected : this.props.selected,
 		}
 	}
-
-	componentWillReceiveProps (nextProps) {
-		this.setState({
-		  modalVisible : false,
-		  defaultText : nextProps.defaultText,
-		  selected : nextProps.selected,
-		})
```
Comment (technology-studio-apiary) [inline lib/select.js:57]: This introduce the original problem back.

**Comment by technology-studio-apiary** [comment]
please remove your last commit e57fc4c, and merge with #20

**Comment by gs-akhan** [comment]
@stoffern  Should I merge this ?

**Comment by st0ffern** [comment]
@gs-akhan wait until we get this sorted out.. 
@technology-studio-apiary or @rostislav-simonik is the same person and wants to fix it i guess..

The problem is:
- when you set a `defaultText` to the `Select` in example `TodoComponent`
- Change a `Select` dropdown..
- Change a state in the `TodoComponent` will trigger `Select` to reset.

**Comment by gs-akhan** [comment]
Alright. @technology-studio-apiary  @rostislav-simonik  Please figure it out and make a PR. I will be more than happy to merge ..
Btw Anyone wants Admin rights to the repo ? I am happy to add you guyz.

**Comment by st0ffern** [comment]
I would love to help 😉

**Comment by gs-akhan** [comment]
@stoffern  There you go !! Go for the glory !! :)

**Comment by technology-studio-apiary** [comment]
Ok, hello guys. I have looked more into the code. And some questions have risen. Imagine situation where parent component is mount I want after that to change currently selected value. Should I call setSelectedText ? Isn't against reactive programming ? I'd like to expect that there will be prop like selectedText which will be reflected through [componentWillReceiveProps](https://developmentarc.gitbooks.io/react-indepth/content/life_cycle/update/component_will_receive_props.html)
https://developmentarc.gitbooks.io/react-
and in case it's empty then content of defaultText is taken. @stoffern @gs-akhan, please let me now how to proceed because proposed change will harm backward compatibility, but it's the proper way how the  select component should work. 

I have prepared example without without introducing setSelectedText prop, and it requires following change in select.js

```
componentWillReceiveProps(nextProps) {
	this.setState({
		defaultText: nextProps.defaultText
	})
}
```
Example: 
```
/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';
import R from 'ramda'

import {
  AppRegistry,
  StyleSheet,
  Text,
  Button,
  View,
} from 'react-native';

import { Select, Option } from './react-native-chooser'

export default class AwesomeProject extends Component {

  constructor(props) {
    super(props)
    this.state = {
      selectedeText: null,
      selected: null,
      farms: [
        {
          id: 1,
          name: 'Nice'
        },
        {
          id: 2,
          name: 'Small'
        },
        {
          id: 3,
          name: 'Little'
        }
      ]
    }
  }

  onSelect = (id) => {
    this.setState({selected: id})
  }

  onChangeSelection = () => {
    this.setState({selected: 2})
  }

  render() {

    return (
      <View style={styles.container}>
        <Select
          onSelect = {this.onSelect}
          defaultText  = {this.state.selected && R.propOr(null, 'name', R.find(R.propEq('id', this.state.selected))(this.state.farms))|| 'Choose'}
          >
          {
            this.state.farms.map((farm, key) =>
              <Option value={farm.id} key={key}>{farm.name}</Option>
            )
          }
        </Select>
        <Button
          onPress={this.onChangeSelection}
          title="Change selected"
        />
        <Button
          onPress={() => {this.forceUpdate()}}
          title="Force update component"
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  }
});

AppRegistry.registerComponent('AwesomeProject', () => AwesomeProject);

```

**Comment by gs-akhan** [comment]
@technology-studio-apiary  hi there. 
I understand the situation where the parent component has mounted. In that case we already have to props. 
`defaultText` and `selected` when passed will set appropriate things into the component right ?
Correct me if I got this wrong . 

Thanks

**Comment by technology-studio-apiary** [comment]
Sorry, I'm little bit confused with your comment, what components are you referencing. Not sure what should I confirm, please reformulate. 

Let me explain problem. Your select component reuses same name defaultText for internal state variable. It can be confusing for the reader because it really references currentText. That would be better name for that state variable. But this is not issue. Issue comes when I want to change dynamically currently selected text after select component (not parent component of select component) has been mounted, so in that case the only option is to call setSelectedText. It works but it's not done in "reactive" way. 

Alternative to setSelectedText is to introduce new prop selectedText for select component or to adapt defaultText to behave as selectedText prop, the way how I have shown in example above. Both approaches require update of select's internal state through componentWillReceiveProps.

**Comment by technology-studio-apiary** [comment]
I can create new pull request, which breaks backward compatibility and new major version of component will be necessary (2.x.x) so everybody who used this component must migrate their code to new logic or stick with version 1.x.x . So question is what would you like to do. To keep the current logic or upgrade?

**Comment by gs-akhan** [comment]
@technology-studio-apiary  Let me try it out locally here in order to validate what you've said. 
Also yeah if we have have make a change then lets do that in 2.x

Also we need to have add few more changes in v2 like 

1. Accepting <View /> as options, so users can add icons in the options.
2. Multiple selections.
3. Positioning the selection from top or bottom of screen. 

Thanks

**Comment by st0ffern** [comment]
closing this in favor for v2

**Pull request closed without merging**